### PR TITLE
fix(workflows): Workflows fixes and new checks

### PR DESCRIPTION
**Pull request by craciunoiuc**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR includes several new fixes and checks:
 - run workflows when they change
 - fix inconsistent syntax for workflows
 - add workflow to lint workflows
 - fix linting reported by the new workflow

### CODE REVIEW

File: .github/workflows/actcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          submodules: recursive
+
+      - name: Install action linter
+        run: |
+          mkdir -p ~/bin
+          curl -sL https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash | bash -s -- latest ~/bin
+
+      - name: Check that all workflows are valid
+        run: |
+          set -xe
```
Comment (antoineco) [inline .github/workflows/actcheck.yaml:26]: nit: `set -e` is the default, it's not necessary to set it again.

**Review by antoineco — APPROVED** [review]
Reviewed-by: Antoine Cotten <antoine@unikraft.io>

File: .github/workflows/actcheck.yaml (hunk 1 above)
Comment (craciunoiuc) [reply .github/workflows/actcheck.yaml:26]: didn't know this, nice

Comment (craciunoiuc) [reply .github/workflows/actcheck.yaml:26]: I see you approved, I guess we can leave it in, doesn't hurt :sweat_smile:

Comment (antoineco) [reply .github/workflows/actcheck.yaml:26]: I approved with a nit pick, didn't want to block the PR on that.

**Review by nderjung — APPROVED** [review]
Thanks!

Approved-by: Alexander Jung <alex@unikraft.io>

**Pull request merged**